File: Programming/Java/TwoSum/twoSum.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);
        System.out.println("Enter the array length: ");
        int n = sc.nextInt();
        int arr[] = new int[n];
        System.out.println("Enter array elements (elements should be *Non-Decresing* order): ");
        for (int i = 0; i < n; i++) {
            arr[i] = sc.nextInt();
        }
        System.out.println("Enter a target sum: ");
        int target = sc.nextInt();

        twoSum(arr, n, target);

    }

    public static void twoSum(int[] arr, int n, int target) {
        int left = 0, right = n - 1;
        ArrayList<Integer> res = new ArrayList<Integer>(3);
        while (left < right) {
            int sum = arr[left] + arr[right];
            if (sum == target) {
                res.add(left);
                res.add(right);
                break;
            } else if (sum > target) {
                right--;
            } else {
                left++;
            }
        }
        for (int i = 0; i < res.size(); i++) {
            int ind = res.get(i);
            System.out.print(ind + 1 + " ");
        }
    }
}

File: Programming/Kotlin/TwoSum/TwoSum.kt
package com.example.testapp

import java.util.*

fun main(){
    val numbers = arrayListOf<Int>()
    val output = arrayListOf<Int>()
    val scanner = Scanner(System.`in`)

    println("Enter the size of the list")
    val listSize = scanner.nextInt()

    println("Enter the $listSize numbers in a non-decreasing sorted manner")
    for (i in 0 until listSize) numbers.add(scanner.nextInt())

    println("Enter the target number")
    val targetNum = scanner.nextInt()

    outerLoop@
    for (i in 0 until listSize){
        for (j in i+1 until listSize){
            if (numbers[i]+numbers[j] == targetNum){
                output.addAll(arrayListOf(i+1,j+1))
                break@outerLoop
            }
        }
    }

    if (output.isNotEmpty()) println("[${output[0]},${output[1]}]")
    else println("The list does not contain the required numbers to satisfy the condition")
}